tests/functional/test_patient_create.py

```diff
@@ -0,0 +1,159 @@
+        "X-Correlation-ID": str(uuid.uuid4()),
+        "Authorization": f'Bearer {access_token}'
+    }
+
+    role_id = get_role_id_from_user_info_endpoint(access_token, identity_service_base_url)
+    headers.update({"NHSD-Session-URID": role_id})
+
+    return headers
+
+
+# SUPPORTING FUNCTIONS-------------------------------------------------------------------------------------------
+async def _create_patient(session, headers, url, body):
+    details = {'request_time': datetime.datetime.now(datetime.timezone.utc)}
+
+    async with session.post(url=url, headers=headers, data=body) as resp:
+        status = resp.status
+        headers = resp.headers
+        text = await resp.text()
+        try:
+            json_obj = json.loads(text)
+        except ValueError:
+            json_obj = resp.json()
+        response_dict = json_obj
+        details['status'] = status
+        details['response'] = response_dict
+        details['response_time'] = parser.parse(headers['Date'])
+        return details
+
+
+async def _create_all_patients(headers, url, body, loop, num_patients):
+    conn = aiohttp.TCPConnector(limit=3)
+    async with aiohttp.ClientSession(connector=conn, loop=loop) as session:
+        results = await asyncio.gather(
+            *[_create_patient(session, headers, url, body) for _ in range(num_patients)],
+            return_exceptions=True
+        )
+        return results
+
+
+# STEPS----------------------------------------------------------------------------------------------------------
+@pytest.mark.asyncio
+@when("I post to the Patient endpoint more than 5 times per second", target_fixture='post_results')
+def post_patient_multiple_times(healthcare_worker_auth_headers: dict, pds_url: str) -> list:
+    # firing 30 requests in 5 seconds should trigger the spike arrest policy
+    # about 5 times...
+    patients_to_create = 30
+    target_request_time = 5
+
+    url = f'{pds_url}/Patient'
+    body = json.dumps({"nhsNumberAllocation": "Done"})
+
+    loop = asyncio.new_event_loop()
+    results = loop.run_until_complete(
+        _create_all_patients(healthcare_worker_auth_headers, url, body, loop, patients_to_create)
+    )
+    request_times = [x['request_time'] for x in results]
+    request_times.sort()
+    elapsed_time_req = request_times[-1] - request_times[0]
+    assert elapsed_time_req.seconds == 0
+
+    response_times = [x['response_time'] for x in results]
+    response_times.sort()
+    elapsed_time_res = response_times[-1] - response_times[0]
+
+    # we fired requests at or faster than the expected rate
+    assert elapsed_time_res.seconds <= target_request_time
+
+    return results
+
+
+@then("I get a mix of 400 and 429 HTTP response codes")
+def assert_expected_spike_arrest_response_codes(post_results):
+    successful_requests = [x for x in post_results if x['status'] == 400]
+    spike_arrests = [x for x in post_results if x['status'] == 429]
+    assert len(spike_arrests) >= 5
```

Comment:
Getting a mix just means there's some, right? Why 5? Makes this fixture less reusable?

Reply:
@ChrisClarke-NHSD I don't intend to reuse this step - if we ever wanted to reuse it then we could change it, but for now I think does what it needs to do. For the way this test is configured, we expect at least 5.

This step will have to change once the endpoint behaviour changes too - we currently expect a 400 but one day we might expect it to work and return a 200 (and when that day comes we probably don't actually want to create loads of new nhs numbers in spine every time we run tests, so maybe we don't ever want to try creating a patient for real? But we'll get there when the moment comes).